tests/cpgrid/LgrChecks.hpp

```diff
@@ -168,61 +216,94 @@ void Opm::checkEqMinMaxGlobalCellLevelZeroAndLeaf(const std::vector<int>& global
 }
 
 void Opm::checkFatherAndSiblings(const Dune::cpgrid::Entity<0>& element,
-                                 int level,
+                                 int preAdaptMaxLevel,
                                  double expected_total_children,
                                  const Dune::CpGrid& grid)
 {
-    BOOST_CHECK( element.father().level() == 0); // If there is no nested refinement
-    BOOST_CHECK( element.father().isLeaf() == false); // Father vanished during refinement.
-    BOOST_CHECK( element.father() == element.getOrigin() );
+    const auto& father = element.father();
+    const auto& fatherLevelData = grid.currentData()[father.level()];
 
-    auto itFather = element.father().hbegin(grid.maxLevel());
-    const auto endItFather = element.father().hend(grid.maxLevel());
-    // If itFather != endItFather and !element.father().isLeaf() (if dristibuted_data_ is empty).
-    BOOST_CHECK( itFather != endItFather );
-    Opm::checkAverageCenterAndVolume(itFather, endItFather, expected_total_children);
+    if (preAdaptMaxLevel) {
+        BOOST_CHECK( father.level() <= preAdaptMaxLevel );
+    }
+    else {
+        BOOST_CHECK_EQUAL( father.level(), 0 ); // If there is no nested refinement
+    }
```

Comment:
Do we need if-else? If preAdaptMaxLevel is false then it is 0 anyway and the first check should suceed.

Reply:
Good point! Thanks!